<launch>
  <arg name="cheetah_namespace" default="cheetah" />


  <node name="sim_publisher_node" pkg="cheetah_inekf_ros" type="sim_publisher_node" output="screen" required="true" >
        <param name="is_listening_sim" value="true" />
        <param name="is_publishing_lcm" value="true" />
        <param name="publish_rate" value="100"/>
        <param name="joint_state_lcm_topic" value="leg_control_data"/>
        <param name="sim_lcm_topic" value="simulator_state"/>
        <param name="imu_lcm_topic" value="microstrain"/>
        <param name="sim_contact_force_threshold" value="40.0"/>
        <param name="encoder_std" value="0.05"/>
        <param name="kinematic_prior_orientation_std" value="0.01"/>
        <param name="kinematic_prior_position_std" value="0.01"/>
    </node>

      <!-- Run Invariant EKF -->
    <node name="inekf" pkg="inekf_ros" type="inekf_node" output="screen" required="true">
        <rosparam file="$(find cheetah_inekf_ros)/config/inekf/sim_settings.yaml" command="load" />
        <rosparam file="$(find cheetah_inekf_ros)/config/inekf/noise.yaml" command="load" />
        <rosparam file="$(find cheetah_inekf_ros)/config/inekf/prior.yaml" command="load" />
    </node> 

    <!-- Run odom path publisher -->

    <node name="path_publisher" pkg="cheetah_inekf_ros" type="path_publisher_node" required="true">
      <param name="pose_topic" value="/cheetah/pose"/>
      <param name="path_topic" value="/cheetah/path"/>
      <param name="publish_rate" value="1"/>
      <param name="pose_skip" value="100"/>
    </node>

    <node name="path_publisher_sim" pkg="cheetah_inekf_ros" type="path_publisher_node" required="true" >
      <param name="pose_topic" value="/sim_publisher_node/sim_pose"/>
      <param name="path_topic" value="/cheetah/path"/>
      <param name="publish_rate" value="1"/>
      <param name="pose_skip" value="100"/>
    </node>




    
  </launch>
